### PR TITLE
Add additional checks to metadata upload script

### PR DESCRIPTION
**Pull request by jbarrett**
##### Description :

Additional validation and upload stages for metadata upload

- JSON validity
- bzip file validity
- Atomic set-and-move depending on s3cmd success

##### References issues / PRs:

#

##### Who should be informed of this change?

@jkv @kablamo 

##### Checklist :
- [ ] Back end tests (perl, scripts)
- [ ] Front end tests (js, integration)
- Browser verification
    - [ ] IE
    - [ ] Chrome
    - [ ] Firefox
    - [ ] Safari
    - [ ] Opera 
- Mobile verification
    - [ ] iOS
    - [ ] Android

### CODE REVIEW

File: script/ddgc_upload_metadata.sh [hunk 1]
```diff
@@ -25,8 +26,14 @@ NEW_MODIFIED_TIME=0
   NEW_MODIFIED_TIME=$(/usr/bin/stat --format=%Y $DDGC_REPO_JSON_OUT)
 
 if [[ ( $DOWNLOAD_EXIT_CODE -eq 0 && $NEW_MODIFIED_TIME -gt $LAST_MODIFIED_TIME ) || ( $DOWNLOAD_EXIT_CODE -eq 0 && $FORCE_UPLOAD -eq 1 ) ]] ; then
-  bzip2 --best --force --keep $DDGC_REPO_JSON_OUT
+  json_xs < $DDGC_REPO_JSON_OUT > /dev/null
```
Comment (kablamo) [inline script/ddgc_upload_metadata.sh]: does this exist in the $PATH?  I mean I'm guessing it does. I just wanted to double check that.

Comment (jbarrett) [inline script/ddgc_upload_metadata.sh]: Yes, should be once the perlbrew config is sourced.

File: script/ddgc_upload_metadata.sh [hunk 2]
```diff
@@ -25,8 +26,14 @@ NEW_MODIFIED_TIME=0
...
   [ "$?" == "0" ] && \
-    s3cmd -P -m 'application/x-bzip2' --add-header='Content-Encoding: bzip2' --force -c /usr/local/etc/s3cmd/ddgc-metadata.s3cfg put $DDGC_REPO_BZIP2_OUT $DDGC_REPO_JSON_S3URL > /dev/null
+    bzip2 --best --force --keep $DDGC_REPO_JSON_OUT
+    [ "$?" == "0" ] && \
+      bzip2 -t $DDGC_REPO_BZIP2_OUT
+      [ "$?" == "0" ] && \
+        s3cmd -P -m 'application/x-bzip2' --add-header='Content-Encoding: bzip2' --force -c /usr/local/etc/s3cmd/ddgc-metadata.s3cfg put $DDGC_REPO_BZIP2_OUT $DDGC_REPO_JSON_S3URL_TMP > /dev/null
+        [ "$?" == "0" ] && \
+          s3cmd -c /usr/local/etc/s3cmd/ddgc-metadata.s3cfg mv $DDGC_REPO_JSON_S3URL_TMP $DDGC_REPO_JSON_S3URL > /dev/null
```
Comment (jbarrett) [inline script/ddgc_upload_metadata.sh]: Do we need `--force` here, since the destination will likely exist?

Comment (kablamo) [inline script/ddgc_upload_metadata.sh]: seems like it

**Comment by jkv** [comment]
`json_xs` will exist on the path because we do `source /etc/profile.d/perlbrew.sh` before running the script.

**Pull request merged**